### PR TITLE
[ports] add support expressions

### PR DESCRIPTION
**Pull request by autoantwort**
Tested them on my arm mac and they don't support arm.

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

File: ports/apsi/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "apsi",
   "version-semver": "0.7.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "APSI is a research library for asymmetric private set intersection.",
   "homepage": "https://github.com/microsoft/APSI",
-  "supports": "static",
+  "supports": "static & !arm",
```
Comment (ras0219-msft) [inline ports/apsi/vcpkg.json:7]: It does look like the library attempts to support arm & arm64 (https://github.com/microsoft/APSI/blob/main/common/apsi/fourq/FourQ.h#L83), so I don't think `!arm` is right here.

I did a little bit of investigation and discovered the failure might be due to the flatbuffers dependency, which I've attempted to address https://github.com/microsoft/vcpkg/pull/23067.

Could you leave this change off of this PR?

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/embree2/vcpkg.json [hunk 2]
```diff
@@ -1,9 +1,10 @@
 {
   "name": "embree2",
   "version-semver": "2.17.7",
-  "port-version": 3,
+  "port-version": 4,
   "description": "High Performance Ray Tracing Kernels.",
   "homepage": "https://github.com/embree/embree",
+  "supports": "!arm",
```
Comment (cenit) [inline ports/embree2/vcpkg.json:7]: this is not removed from baseline, so it was working on arm on windows?

Comment (autoantwort) [reply ports/embree2/vcpkg.json:7]: No, the dependency tbb already does not work on arm.

File: ports/apsi/vcpkg.json (hunk 1 above)
Comment (autoantwort) [reply ports/apsi/vcpkg.json:7]: On my arm macbook I get the following error:
```
CMake Error at common/apsi/fourq/CMakeLists.txt:30 (add_subdirectory):
  add_subdirectory given source "arm64" which is not an existing directory.
```
That indicated it does not support arm, at least not on mac.

Comment (autoantwort) [reply ports/apsi/vcpkg.json]: Ok I have removed apsi from this PR. @JonLiu1993 you can mark this as `reviewed` again :)

File: ports/embree2/vcpkg.json (hunk 2 above)
Comment (cenit) [reply ports/embree2/vcpkg.json:7]: which is absolutely not important to mark this downstream as not supporting arm…

Comment (autoantwort) [reply ports/embree2/vcpkg.json:7]: But is does not support arm on macos: 
[install-arm64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/8061802/install-arm64-osx-dbg-out.log)

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Now that #23067 is merged, can you try again and let us know how it goes?

**Comment by autoantwort** [comment]
apsi still does not build on arm64-osx. So I added the support expression back for macOS

**Review by JonLiu1993 — APPROVED** [review]

**Comment by vicroms** [comment]
Trigger CLA check again

**Pull request merged**